azero/contracts/most/lib.rs

```diff
@@ -156,6 +156,8 @@ pub mod most {
         gas_price_oracle: Option<AccountId>,
         /// Is the bridge in a halted state
         is_halted: bool,
+        /// Wrapped ethereum (azero) address. Necessary to perform bridging weth(azero) -> ether(eth).
+        weth: Option<AccountId>,
```

Comment:
Question: shouldn't this be in a separate `Lazy` to ensure a smooth upgrade on testnet?

Reply:
You are correct. An upgrade would break the `data: Lazy<Data,...>` because `weth` is not initialized. It should be moved to a separate `Lazy`.

Reply:
you could add a migration that copies all the old data and sets the new field:
```
            if let Some(OldData { x, y, .. }) = get_contract_storage(&123)? {               
                self.data.set(&Data { x: x, y: y, z: weth_address });
                return Ok(());
            }
```

Reply:
This way we need to keep the definition of the "`OldData`" just for the process of migration. I agree there is no need for the `set_weth(...)` but instead we need `migrate(...)`. This introduces more complexity imho.